lib/membrane_live_audio_mixer.ex

```diff
@@ -0,0 +1,217 @@
+    mode: :pull,
+    availability: :on_request,
+    demand_unit: :bytes,
+    accepted_format:
+      any_of(
+        %RawAudio{sample_format: sample_format}
+        when sample_format in [:s8, :s16le, :s16be, :s24le, :s24be, :s32le, :s32be],
+        Membrane.RemoteStream
+      ),
+    #
+    options: [
+      offset: [
+        spec: Time.non_neg_t(),
+        default: 0,
+        description: "Offset of the input audio at the pad."
+      ]
+    ]
+
+  @impl true
+  def handle_init(_ctx, %__MODULE__{stream_format: stream_format} = options) do
+    if options.native_mixer && !options.prevent_clipping do
+      raise("Invalid element options, for native mixer only clipping preventing one is available")
+    else
+      {:ok, live_queue_state} = LiveQueue.init(stream_format)
+
+      state =
+        options
+        |> Map.from_struct()
+        |> Map.put(:mixer_state, initialize_mixer_state(stream_format, options))
+        |> Map.put(:last_ts_sent, 0)
+        |> Map.put(:live_queue, live_queue_state)
+
+      {[], state}
+    end
+  end
+
+  @impl true
+  def handle_playing(_context, %{stream_format: %RawAudio{} = stream_format} = state) do
+    {[
+       stream_format: {:output, stream_format},
+       start_timer: {:timer, Membrane.Time.milliseconds(100)}
+     ], state}
+  end
+
+  def handle_playing(_context, %{stream_format: nil} = state) do
+    {[start_timer: {:timer, Membrane.Time.milliseconds(100)}], state}
+  end
+
+  @impl true
+  def handle_start_of_stream(
+        Pad.ref(:input, pad_id) = pad,
+        context,
+        %{live_queue: live_queue} = state
+      ) do
+    offset = context.pads[pad].options.offset
+
+    {:ok, new_live_queue} = LiveQueue.add_queue(pad_id, offset, live_queue)
+
+    {[], %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_end_of_stream(Pad.ref(:input, pad_id), context, %{live_queue: live_queue} = state) do
+    {:ok, new_live_queue} = LiveQueue.remove_queue(pad_id, live_queue)
+
+    actions =
+      if all_streams_ended?(context) do
+        [{:end_of_stream, :output}]
+      else
+        []
+      end
+
+    {actions, %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_process(
+        Pad.ref(:input, pad_id),
+        buffer,
+        _context,
+        %{live_queue: live_queue} = state
+      ) do
+    {:ok, new_live_queue} = LiveQueue.add_buffer(pad_id, buffer, live_queue)
+
+    {[], %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_tick(_timer_id, _context, state) do
+    {payload, state} = mix(Membrane.Time.milliseconds(100), state)
+    {[buffer: {:output, %Buffer{payload: payload}}], state}
+  end
+
+  @impl true
+  def handle_stream_format(_pad, stream_format, _context, %{stream_format: nil} = state) do
+    state = %{state | stream_format: stream_format}
+    mixer_state = initialize_mixer_state(stream_format, state)
+
+    {[stream_format: {:output, stream_format}, redemand: :output],
```

Comment:
I assume that demands are going to be implemented

Reply:
it will use auto demands